aws/common/s3.tf

```
@@ -249,6 +249,51 @@ resource "aws_s3_bucket" "document_bucket" {
   }
 }
 
+resource "aws_s3_bucket" "scan_files_document_bucket" {
+  bucket = "notification-canada-ca-${var.env}-document-download-scan-files"
+  acl    = "private"
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+
+  # expire linked files after 7 days
+  lifecycle_rule {
+    id      = "tf-s3-lifecycle-linked-files"
+    enabled = true
+
+    expiration {
+      days = 7
+    }
+  }
+
+  # Expire files attached directly to emails after a few days.
+  # Those are stored in a `tmp/` folder.
+  # See https://github.com/cds-snc/notification-document-download-api
+  lifecycle_rule {
+    id      = "tf-s3-lifecycle-attached-files"
+    enabled = true
+    prefix  = "tmp/"
+
+    expiration {
+      days = 3
+    }
+  }
+
+  #tfsec:ignore:AWS077 - Versioning is not enabled
+  logging {
+    target_bucket = aws_s3_bucket.document_bucket_logs.bucket
+  }
+
+  tags = {
+    CostCenter = "notification-canada-ca-${var.env}"
+  }
+}
+
```

If you wanted, you could use the CDS `S3` module to create the bucket like so:
```suggestion
module "scan_files_document_bucket" {
  source      = "github.com/cds-snc/terraform-modules?ref=v3.0.20//S3"
  bucket_name = "notification-canada-ca-${var.env}-document-download-scan-files"
  lifecycle_rule = [
    {
      id      = "tf-s3-lifecycle-linked-files"
      enabled = true
      expiration {
        days = 7
      }
    },
    {
      id      = "tf-s3-lifecycle-attached-files"
      enabled = true
      prefix  = "tmp/"
      expiration {
        days = 3
      }
    }
  ]
  billing_tag_value = "notification-canada-ca-${var.env}"
  logging = {
    target_bucket = aws_s3_bucket.document_bucket_logs.bucket
  }
}
```

---

Is encryption automatically handled with this module? If yes, default encryption? i.e. no KMS key?

---

Yup, exactly so - it defaults to `AES256` if a KMS key isn't specified:
https://github.com/cds-snc/terraform-modules/blob/8c3bdffea8b6f8dc8244e1a5f1a8e0141cd4fd28/S3/main.tf#L66-L75

---

If security team is good with it, we are good with it. We should have @ascheid looped in for reviewing and the [ADR a well](https://github.com/cds-snc/notification-adr/pull/30).